cmd/cloudinit/cloudinit.go

```diff
@@ -229,7 +230,9 @@ func saveCloudConfig() error {
 		}
 	}
 
-	userData := string(userDataBytes)
+	env := initialize.NewEnvironment("", "", "", "", metadata)
+	userData := env.Apply(string(userDataBytes))
+	userDataBytes = []byte(userData)
```

Comment:
Just to make myself clear about the unit test I requested: please, turn these lines into a function and add a test illustrating what it does. `saveCloudConfig()` as a whole is, frankly, not unit-testable as is.